iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -25,6 +25,8 @@ namespace iox
 {
 namespace cxx
 {
+// NOLINTJUSTIFICATION Not all elements in the array shall be initialized
```

Comment:
I would state here that the upcoming uninitialized array will solve this issue. Here and in the remaining uninitialized warning suppressions.